src/main/resources/META-INF/rewrite/java-version-25.yml

```diff
@@ -0,0 +1,105 @@
+#
+# Copyright 2025 the original author or authors.
+# <p>
+# Licensed under the Moderne Source Available License (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://docs.moderne.io/licensing/moderne-source-available-license
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.UpgradeToJava25
+displayName: Migrate to Java 25
+description: >-
+  This recipe will apply changes commonly needed when migrating to Java 25. This recipe will also replace deprecated API
+  with equivalents when there is a clear migration strategy. Build files will also be updated to use Java 25 as the
+  target/source and plugins will be also be upgraded to versions that are compatible with Java 25.
+tags:
+  - java25
+recipeList:
+  - org.openrewrite.java.migrate.RemoveSecurityManagerAPI
+
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.RemoveSecurityManagerAPI
+displayName: Remove Security Manager API
+description: The Security Manager API is unsupported in Java 24. This recipe will remove the API and any usages of it.
+tags:
+  - java25
+  - security
+  - deprecation
+preconditions:
+  - org.openrewrite.java.migrate.search.FindJavaVersion:
+      version: 24
+recipeList:
```

Comment:
Alternative to the suggestion below; pick this one or the other one, but not both.
```suggestion
```

Reply:
The Leaf recipes (`RemoveSecurityPolicy`, `RemoveSecurityManager` and `AccessController`) should guard that they are only executed against Java 24 as these changes are only needed upgrading to Java 24. Check on `RemoveSecurityManagerAPI` is not needed, agree. And `RemoveSecurityManagerAPI` is two hands full of clicks in Recipe Builder.